### PR TITLE
[CARBONDATA-3889] Cleanup duplicated code in carbondata-core module

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
 There are duplicated code in carbondata-core module
 
 ### What changes were proposed in this PR?
1. Cleanup duplicated code in carbondata-core module
2. Fix the typo
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3480/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1738/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1752/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3494/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3498/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1756/

File: core/src/main/java/org/apache/carbondata/core/index/dev/expr/AndIndexExprWrapper.java [hunk 1]
```diff
@@ -47,25 +47,20 @@ public AndIndexExprWrapper(IndexExprWrapper left, IndexExprWrapper right,
...
-          throws IOException {
-    List<ExtendedBlocklet> leftPrune = left.prune(distributable, partitionsToPrune);
-    List<ExtendedBlocklet> rightPrune = right.prune(distributable, partitionsToPrune);
+      List<PartitionSpec> partitionsToPrune) throws IOException {
+    return and(left.prune(distributable, partitionsToPrune),
+        right.prune(distributable, partitionsToPrune));
+  }
+
+  private List<ExtendedBlocklet> and(List<ExtendedBlocklet> leftPrune,
+      List<ExtendedBlocklet> rightPrune) {
```
Comment (kevinjmh) [inline core/src/main/java/org/apache/carbondata/core/index/dev/expr/AndIndexExprWrapper.java:63]: `pruneBlocklets` also can apply to this method too

File: core/src/main/java/org/apache/carbondata/core/datastore/chunk/impl/FixedLengthDimensionColumnPage.java [hunk 2]
```diff
@@ -136,15 +131,29 @@ public int fillVector(ColumnVectorInfo[] vectorInfo, int chunkIndex) {
           } else if (dataType == DataTypes.LONG) {
             vector.putLong(vectorOffset++, (long) valueFromSurrogate);
           } else {
-            throw new IllegalArgumentException("unsupported data type: " +
-                columnVectorInfo.directDictionaryGenerator.getReturnType());
+            throw new IllegalArgumentException(
+                "unsupported data type: " + columnVectorInfo.directDictionaryGenerator
+                    .getReturnType());
```
Comment (kevinjmh) [inline core/src/main/java/org/apache/carbondata/core/datastore/chunk/impl/FixedLengthDimensionColumnPage.java:136]: no need to change this

Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/datastore/chunk/impl/FixedLengthDimensionColumnPage.java]: reverted

File: core/src/main/java/org/apache/carbondata/core/index/dev/expr/AndIndexExprWrapper.java (hunk 1 above)
Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/index/dev/expr/AndIndexExprWrapper.java]: done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3503/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1761/

**Comment by kevinjmh** [comment]
LGTM

**Pull request closed without merging**